# 1 задача
#### 1.1. Скачать файл в уроке

#### 1.2. Считать данные с помощью pandas




In [6]:
import pandas as pd

In [7]:
df = pd.read_csv('/Users/anastasiaolegovna/Downloads/GeekBrains/Jupiter_Phyton/2/kc_house_data.csv')

#### 1.3. Вывести на экран первые 5 строк

#### 1.4. Посмотреть на описание признаков и на их содержание


In [8]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

- **id** - Уникальный ID для каждого дома
- **date** - Дата продажи дома
- **price** - Стоимость продажи дома
- **bedrooms** - Кол-во спален
- **bathrooms** - Кол-во ванных комнат (0.5 - туалет без душа)
- **sqft_living** - Кв. метры жилые
- **sqft_lot** - Кв. метры общие
- **floors** - Кол-во этажей
- **waterfront** - Есть набержная или нет
- **view** - Значение от 0 до 4 насколько хороший вид
- **condition** - Значение от 1 до 5 насколько хорошее состояние
- **grade** - Значение от 1 до 13, где 1-3 плохая конструкция здания и дизайн, 7 - средний уровень конструкции и дизайна, 11-13 - высокое качество конструкции и дизайна
- **sqft_above** - Кв. метры дома, которые находятся выше земли
- **sqft_basement** - Кв. метры дома, которые находятся ниже земли
- **yr_built** - Год постройки дома
- **yr_renovated** - Год ремонта дома
- **zipcode** - Индекс
- **lat** - Широта
- **long** - Долгота
- **sqft_living15** - Кв. метры жилой площади у 15 соседей
- **sqft_lot15** - Кв. метры общей площади у 15 соседей

# 2 задача
#### 2.1. Проведите первичный анализ данных

#### 2.2. Изучите типы данных

In [19]:
df.dtypes

id                 int64
date              object
price            float64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront         int64
view               int64
condition          int64
grade              int64
sqft_above         int64
sqft_basement      int64
yr_built           int64
yr_renovated       int64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object

In [12]:
df.shape

(21613, 21)

#### 2.3. Найдите количество пропущенных ячеек в данных



In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

#### 2.4. Посчитайте основные статистики по всем признакам и поизучайте их



In [18]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [16]:
df.describe(include=['object'])

date
count             21613
unique              372
top     20140623T000000
freq                142

#### 2.5. Пишите выводы

###### 1). Типы данных
dtypes: 
- float64(5), 
- int64(15), 
- object(1)

###### 2).  Количество пропущенных ячеек в данных 
0, то есть у нас есть вся информация 

###### 3). Посчитайте основные статистики по всем признакам и поизучайте их

- Количество строк (объектов для сравнения по признакам): 21613
- Количество столбцов (признаков):  21


- 2014.06.23 - в этот день совершено максимальное кол-во сделок по продаже дома - 142 продажи
- данные охватывают 372 дня

# 3 задача
Ответьте на несколько вопросов
#### 3.1. В каком диапазоне изменяются стоимости недвижимости?




In [27]:
df['price'].min(), df['price'].max()  # диапазон стоимости недвижимости

(75000.0, 7700000.0)

#### 3.2. Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

- **sqft_living** - Кв. метры жилые
- **sqft_lot** - Кв. метры общие

In [31]:
df['sqft_living'].sum()  # жилая площадь по всем домам

44952873

In [32]:
df['sqft_lot'].sum()  # вся площадь по всем домам

326506890

In [36]:
(df['sqft_living'].sum() / df['sqft_lot'].sum() * 100).round(2)

13.77

#### 3.3. Как много домов с разными этажами в данных?

- **floors** - Кол-во этажей

In [59]:
df['floors'].value_counts()  

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

#### 3.4. Насколько хорошие состояния у домов в данных?

- **condition** - Значение от 1 до 5 насколько хорошее состояние


In [62]:
df['condition'].value_counts()  

# наибольшее количество домов со среднем состоянием (оценки 3, 4)
# наименьшее количестов домов с наихудшим состоянием (оценка 1)

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

#### 3.5. Найдите года, когда построили первый дом, когда построили последний дом в данных?

- **yr_built** - Год постройки дома


In [66]:
df['yr_built'].min(), df['yr_built'].max()

(1900, 2015)

# 4 задача
Ответьте на несколько вопросов

#### 4.1. Сколько в среднем стоят дома, у которых 2 спальни?

- **bedrooms** - Кол-во спален


In [74]:
(df[df['bedrooms'] == 2]['price'].mean()).round(2)

401372.68

#### 4.2. Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

- **sqft_lot** - Кв. метры общие


In [79]:
(df[df['price'] > 600000]['sqft_lot'].mean()).round(2)

20442.52

#### 4.3. Как много домов коснулся ремонт?

- **yr_renovated** - Год ремонта дома


In [88]:
df[df['yr_renovated'] != 0].shape[0]

914

#### 4.4. Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [96]:
df[df['grade'] > 10]['price'].mean().round(2)  # средняя стоимость домов с оценкой grade выше 10 

1678635.12

In [95]:
df[df['grade'] < 4]['price'].mean()  # средняя стоимость домов с оценкой grade меньше 4

189750.0

In [97]:
df[df['grade'] > 10]['price'].mean().round(2) - df[df['grade'] < 4]['price'].mean()

1488885.12

# 5 задача
Ответьте на несколько вопросов

#### 5.1. Выберите дом клиенту

Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

- **waterfront** - Есть набержная или нет
- **bathrooms** - Кол-во ванных комнат (0.5 - туалет без душа)
- **sqft_basement** - Кв. метры дома, которые находятся ниже земли


In [116]:
df[(df['waterfront'] == 1) & (df['bathrooms'] >= 3) & (df['sqft_basement'] != 0)].shape[0]

41

#### 5.2. Выберите дом клиенту

Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

- **view** - Значение от 0 до 4 насколько хороший вид
- **waterfront** - Есть набержная или нет
- **condition** - Значение от 1 до 5 насколько хорошее состояние
- **yr_built** - Год постройки дома


In [117]:
df_price = df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] > 1980)]['price']

df_price.min(), df_price.max()

(1295000.0, 3000000.0)

#### 5.3. Выберите дом клиенту

Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

- **sqft_basement** - Кв. метры дома, которые находятся ниже земли
- **floors** - Кол-во этажей
- **price** - Стоимость продажи дома
- **condition** - Значение от 1 до 5 насколько хорошее состояние


In [126]:
df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] <= 150000)]['condition'].mean()

3.0

In [119]:
df.sample(9)

id             date      price  bedrooms  bathrooms  \
8510   7882900120  20140507T000000   230000.0         3       2.50   
14367  3333002450  20150122T000000   490000.0         1       1.00   
13832  1930300410  20150304T000000   575000.0         2       1.00   
15720  2782100280  20140529T000000   672500.0         4       2.75   
13895  2500600297  20140814T000000   243500.0         3       3.00   
7680   3378100100  20150317T000000   345000.0         4       1.50   
14885  8673400177  20150402T000000   525000.0         3       3.00   
16906  3445000005  20141117T000000   237600.0         2       1.00   
11233  2655500241  20140814T000000  1699000.0         3       3.25   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
8510          1920      9180     2.0           0     0  ...      8   
14367          850      8050     1.0           0     0  ...      6   
13832         1250      4320     1.0           0     0  ...      7   
15720         2620      6707     2.0           0     0  ...      9   
13895         2110      7794     1.0           0     0  ...      6   
7680          1540      7168     1.0           0     0  ...      7   
14885         1730      1074     3.5           0     0  ...      8   
16906         1370     11584     1.0           0     0  ...      6   
11233         4160     35153     3.0           0     2  ...     12   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
8510         1920              0      1988             0    98055  47.4818   
14367         850              0      1906             0    98118  47.5427   
13832         850            400      1911             0    98103  47.6549   
15720        2620              0      2000             0    98075  47.5965   
13895        2110              0      1981             0    98198  47.4005   
7680         1160            380      1964             0    98055  47.4550   
14885        1730              0      2006             0    98107  47.6692   
16906        1370              0      1950             0    98198  47.4224   
11233        3690            470      2001             0    98040  47.5749   

          long  sqft_living15  sqft_lot15  
8510  -122.231           1930        7252  
14367 -122.288           1590        5180  
13832 -122.352           1520        4320  
15720 -122.038           2590        6530  
13895 -122.293           1330       10044  
7680  -122.198           1540        7176  
14885 -122.392           1370        1185  
16906 -122.293           1330        8012  
11233 -122.214           3290       11533  

[9 rows x 21 columns]